### DISCUSSION TITLE
[Spec]  Namespace MacOS should be macOS

### DISCUSSION THREAD

**Discussion by Suplanus** [dotnet/maui#48] [General]
Before being to late to the party...

The Namespace `System.Maui.ControlGallery.MacOS` should be named `System.Maui.ControlGallery.macOS`.  
I am not a big fan of starting a part of the namespace with lower case, but its the official naming of Apple and its also considered at the namespace `System.Maui.ControlGallery.iOS`.

# Difficulty : [low]

**Comment by billhenn** [comment]
Namespaces in .NET always start with an uppercase letter.  See [https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions) for details.

"MacOS" as the name adheres to that standard, even if the official name of the O/S is "macOS".

**Comment by ratijas** [comment]
So yeah, back on topic. There are such thing as a code style standard. You either respect it, or you end up with an inconsistent messy codebase. "iOS" case is also highly controversial. Take a look into header files of Cocoa/Foundation — there are tons of macros like `MAC_OS_*` and `IOS_*` — despite the marketed product name, code style for macros SHOUTS OUT IN UPPER CASE because that's the way it used to be.

Now, if you going to throw away the coding guidelines for C#, then what's the point of idiomatic wrappers and APIs? How code completion would look like in IDE — imagine having a list of intermixed upper- and lower-cased namespaces, some IDEs sort them by ASCII (uppercase first). Also, what's next — method names with double-colon ":" in their names?

**Comment by Inrego** [comment]
I guess what @Suplanus is saying, is keep it consistent. If the iOS namespace is called iOS, then macOS namespace should be called macOS. If we insist on having first letter upper case, then the iOS namespace should be IOS.

**Comment by billhenn** [comment]
That's true, "iOS" doesn't fit the standard conventions either, even though it visually looks much better than the other possible variants that do fit the convention: "IOS" or "Ios".

**Comment by Inrego** [comment]
I think the reason why you think iOS looks more visually pleasing, is because that's how we're used to seeing it, due to many years of Apple branding it this way.

I do agree that a brand name should be capitalized according to the brand, even when used in namespaces.

But it's not about what I think, or what you think. It's about making a standard and sticking to it.

**Comment by ratijas** [comment]
"IOS" (capitalized) is actually not that bad. In fact, it is a name of Nintendo Wii's operating system. https://en.wikipedia.org/wiki/Wii_system_software#IOS

**Comment by davidortinau** [comment]
@ratijas I have deleted your inappropriate comments above that violate our [Code of Conduct](https://dotnetfoundation.org/about/code-of-conduct). Future violations of the Code of Conduct will result in a permanent ban. I’ve also deleted a comment from @serialseb with his permission that was calling attention to this behavior.

**Comment by ratijas** [comment]
@davidortinau you calling wat?
